homeassistant/components/qnap_qsw/sensor.py

```diff
@@ -178,21 +350,23 @@ def __init__(
         coordinator: QswDataCoordinator,
         description: QswSensorEntityDescription,
         entry: ConfigEntry,
+        type_id: int | None = None,
     ) -> None:
         """Initialize."""
-        super().__init__(coordinator, entry)
+        super().__init__(coordinator, entry, type_id)
+
         self._attr_name = f"{self.product} {description.name}"
-        self._attr_unique_id = (
-            f"{entry.unique_id}_{description.key}_{description.subkey}"
-        )
+        self._attr_unique_id = f"{entry.unique_id}_{description.key}{description.sep_key}{description.subkey}"
```

Comment:
Is this changing any existing sensors?

Reply:
> Is this changing any existing sensors?

Nope, it isn't because sep_key is `_` by default, so it remains the same for the existing sensors.